torch/csrc/jit/script/schema_matching.cpp

```diff
@@ -417,22 +417,84 @@ MatchedSchema matchSchema(
     const SourceRange& loc,
     Graph& graph,
     at::ArrayRef<NamedValue> args,
-    at::ArrayRef<NamedValue> kwargs) {
+    at::ArrayRef<NamedValue> kwargs,
+    const c10::optional<NamedValue>& self) {
   std::stringstream failure_messages;
   if (auto result = tryMatchSchema(
           schema,
           loc,
           graph,
-          c10::nullopt,
           args,
           kwargs,
+          self,
           &failure_messages,
           /*allow_conversions=*/true)) {
     return *result;
   }
   throw ErrorReport(loc) << failure_messages.str();
 }
 
+static std::string prefixLine(
+    const std::string& str,
+    const std::string& prefix) {
+  std::stringstream ss;
+  bool was_newline = true;
+  for (auto c : str) {
+    if (was_newline)
+      ss << prefix;
+    ss.put(c);
+    was_newline = c == '\n';
+  }
+  return ss.str();
+}
+
+std::pair<size_t, MatchedSchema> matchSchemas(
+    const std::vector<const FunctionSchema*>& schemas,
+    const SourceRange& loc,
+    Graph& graph,
+    at::ArrayRef<NamedValue> args,
+    at::ArrayRef<NamedValue> kwargs,
+    const c10::optional<NamedValue>& self,
+    bool render_errors) {
+  TORCH_INTERNAL_ASSERT(schemas.size() > 0);
+  if (schemas.size() == 1) {
```

Comment:
Why on schema size of 1 are we not trying `allow_conversions` ?

Reply:
matchSchema does do conversions. This is avoiding the duplicate work of first trying without conversions, which is only done when there is multiple schema in order to prioritize non-converting matches first.

Reply:
Maybe add comment